packages/react/src/hooks/useSignIn.ts

```diff
@@ -5,9 +5,38 @@ import type { UseSignInReturn } from '@clerk/types';
 import { useIsomorphicClerkContext } from '../contexts/IsomorphicClerkContext';
 import { useAssertWrappedByClerkProvider } from './useAssertWrappedByClerkProvider';
 
-type UseSignIn = () => UseSignInReturn;
-
-export const useSignIn: UseSignIn = () => {
+/**
+ * The `useSignIn()` hook provides access to the [`SignIn`](https://clerk.com/docs/references/javascript/sign-in/sign-in) object, which allows you to check the current state of a sign-in attempt and manage the sign-in flow. You can use this to create a [custom sign-in flow](https://clerk.com/docs/custom-flows/overview#sign-in-flow).
+ *
+ * @example
+ * ### Check the current state of a sign-in
+ *
+ * The following example uses the `useSignIn()` hook to access the [`SignIn`](https://clerk.com/docs/references/javascript/sign-in/sign-in) object, which contains the current sign-in attempt status and methods to create a new sign-in attempt. The `isLoaded` property is used to handle the loading state.
+ *
+ * ```tsx {{ filename: 'src/pages/SignInPage.tsx' }}
+ * import { useSignIn } from '@clerk/clerk-react'
+ *
+ * export default function SignInPage() {
+ *   const { isLoaded, signIn } = useSignIn()
+ *
+ *   if (!isLoaded) {
+ *     // Handle loading state
+ *     return null
+ *   }
+ *
+ *   return <div>The current sign-in attempt status is {signIn?.status}.</div>
+ * }
+ * ```
+ *
+ * @example
+ * ### Create a custom sign-in flow with `useSignIn()`
+ *
+ * The `useSignIn()` hook can also be used to build fully custom sign-in flows, if Clerk's prebuilt components don't meet your specific needs or if you require more control over the authentication flow. Different sign-in flows include email and password, email and phone codes, email links, and multifactor (MFA). To learn more about using the `useSignIn()` hook to create custom flows, see the [custom flow guides](https://clerk.com/docs/custom-flows/overview).
+ *
+ * ```
+ * ```
```

Comment:
This empty code block is on purpose as without it the whole `@example` block would be interpreted as a code block. Only when you manually add a codeblock the text above is treated as such